tests/test_discovery.py

```diff
@@ -1271,8 +1269,8 @@ def test_method_error_checking(self):
             self.assertTrue("unexpected" in str(e))
 
     def _check_query_types(self, request):
-        parsed = urlparse(request.uri)
-        q = parse_qs(parsed[4])
+        parsed = urllib.parse.urlparse(request.uri)
+        q = urllib.parse.parse_qs(parsed[4])
```

Comment:
Now that we are on Python 3, we can use the fact that `urlparse` returns a named tuple:
```suggestion
        q = urllib.parse.parse_qs(parsed.query)
```

Reply:
Fixed in 5b3a1ac